File: payment-service/src/main/java/com/hedvig/paymentservice/serviceIntergration/memberService/MemberServiceImpl.java
package com.hedvig.paymentservice.serviceIntergration.memberService;

import com.hedvig.paymentservice.serviceIntergration.memberService.dto.Member;
import com.hedvig.paymentservice.serviceIntergration.memberService.dto.SanctionStatus;
import java.util.Optional;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestClientResponseException;

@Service
public class MemberServiceImpl implements MemberService {

  private final Logger log = LoggerFactory.getLogger(MemberServiceImpl.class);
  private final MemberServiceClient memberServiceClient;

  public MemberServiceImpl(
      MemberServiceClient memberServiceClient) {
    this.memberServiceClient = memberServiceClient;
  }

  @Override
  public Optional<Member> getMember(String memberId) {
    try {
      ResponseEntity<Member> response = memberServiceClient.getMember(memberId);
      return Optional.of(response.getBody());
    } catch (RestClientResponseException ex) {
      if (ex.getRawStatusCode() == 404) {
        return Optional.empty();
      }
      log.error("Could not find member {} , {}", memberId, ex);
      throw ex;
    }
  }
}

File: payment-service/src/main/java/com/hedvig/paymentservice/serviceIntergration/memberService/MemberServiceImpl.kt
package com.hedvig.paymentservice.serviceIntergration.memberService

import com.hedvig.paymentservice.serviceIntergration.memberService.dto.Member
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.web.client.RestClientResponseException
import java.util.Optional

@Service
class MemberServiceImpl(
    private val memberServiceClient: MemberServiceClient
) : MemberService {

    override fun getMember(memberId: String): Optional<Member> {
        return try {
            val response = memberServiceClient.getMember(memberId)
            Optional.of(response.body!!)
        } catch (ex: RestClientResponseException) {
            if (ex.rawStatusCode == 404) {
                return Optional.empty()
            }
            log.error("Could not find member {} , {}", memberId, ex)
            throw ex
        }
    }

    override fun getPickedLocale(memberId: String): String {
        return memberServiceClient.getPickedLocale(memberId).pickedLocale!!
    }

    companion object {
        private val log = LoggerFactory.getLogger(MemberServiceImpl::class.java)
    }
}
